### PR TITLE
Adds keypad maglock

### PR DESCRIPTION
**Pull request by Karolis2011**
- [x] Makes maglocks mappable
- [x] Add keypad maglock
- [x] Add vars that let make maglocks invincible and not require power.

### CODE REVIEW

**Comment by Karolis2011** [comment]
!review

File: code/game/objects/items/devices/magnetic_lock.dm [hunk 1]
```diff
@@ -49,6 +51,24 @@
 		desc += " It is painted with [department] colors."
 
 	update_icon()
+	var/obj/machinery/door/airlock/newtarget = (locate(/obj/machinery/door/airlock) in get_turf(src))
+	if(newtarget)
+		var/direction = reverse_direction(dir)
+		forceMove(get_step(newtarget.loc, reverse_direction(direction)))
+		if (locate(/obj/machinery/door/airlock) in oview(1, newtarget))
```
Comment (skull132) [inline code/game/objects/items/devices/magnetic_lock.dm]: Why not condense this to a singular `for (var/obj/machinery/door/airlock/A in oview(1, newtarget))` and check the target direction relative to `newtarget` instead?

File: code/game/objects/items/devices/magnetic_lock.dm [hunk 2]
```diff
@@ -402,8 +429,12 @@
 					return
 				if (1 to 4)
 					add_overlay("overlay_deconstruct_[constructionstate]")
+			if(istype(src, /obj/item/device/magnetic_lock/keypad))
```
Comment (skull132) [inline code/game/objects/items/devices/magnetic_lock.dm]: `istype(src, ...)` is an anti-pattern. Do `/obj/item/device/magnetic_lock/keypad/update_overlays()` instead.

Comment (skull132) [reply code/game/objects/items/devices/magnetic_lock.dm]: wait you already do. why is this here then?

**Pull request merged**